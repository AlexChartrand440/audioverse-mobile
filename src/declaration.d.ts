declare module '*.png';
declare module '*.json';
declare module 'react-native-render-html';
declare module 'react-native-video-controls';
declare module 'react-navigation-addon-search-layout';
declare module 'redux-mock-store';
declare module 'react-native-dotenv';
declare module 'enzyme-adapter-react-16';
