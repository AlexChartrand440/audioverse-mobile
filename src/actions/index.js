export const CHANGE_LANGUAGE = 'CHANGE_LANGUAGE'

export const PLAYER_STATE = 'PLAYER_STATE'
export const PLAYBACK_INIT = 'PLAYBACK_INIT'
export const PLAYBACK_TRACK_ID = 'PLAYBACK_TRACK_ID'
export const PLAYBACK_TRACKS = 'PLAYBACK_TRACKS'

export const RESET_AND_PLAY_TRACK = 'RESET_AND_PLAY_TRACK'
export const PLAY_PAUSE = 'PLAY_PAUSE'

export const SKIP_TO_PREVIOUS = 'SKIP_TO_PREVIOUS'
export const SKIP_TO_NEXT = 'SKIP_TO_NEXT'
export const REPLAY = 'REPLAY'
export const FORWARD = 'FORWARD'
export const DOWNLOAD = 'DOWNLOAD'
export const SET_RATE = 'SET_RATE'
export const TRACK_INITIALIZED = 'TRACK_INITIALIZED'
export const PLAYBACK_RATE = 'PLAYBACK_RATE'
export const PLAYBACK_POSITION = 'PLAYBACK_POSITION'

export const SET_BIBLE_VERSION = 'SET_BIBLE_VERSION'
export const BIBLE_VERSION = 'BIBLE_VERSION'
export const BIBLE_BOOK = 'BIBLE_BOOK'
export const BIBLE_CHAPTER = 'BIBLE_CHAPTER'

const REQUEST = 'REQUEST'
const SUCCESS = 'SUCCESS'
const REFRESH = 'REFRESH'
const FAILURE = 'FAILURE'

function createRequestTypes(base) {
  return [REQUEST,  SUCCESS, REFRESH, FAILURE].reduce((acc, type) => {
    acc[type] = `${base}_${type}`
    return acc
  }, {})
}

function action(type, payload = {}) {
  return {type, ...payload}
}

export const BIBLE_BOOKS = createRequestTypes('BIBLE_BOOKS')
export const BIBLE_CHAPTERS = createRequestTypes('BIBLE_CHAPTERS')
export const NEW_RECORDINGS = createRequestTypes('NEW_RECORDINGS')
export const TRENDING_RECORDINGS = createRequestTypes('TRENDING_RECORDINGS')
export const FEATURED_RECORDINGS = createRequestTypes('FEATURED_RECORDINGS')
export const BOOKS = createRequestTypes('BOOKS')
export const BOOK = createRequestTypes('BOOK')
export const STORIES = createRequestTypes('STORIES')
export const STORY = createRequestTypes('STORY')
export const PRESENTERS = createRequestTypes('PRESENTERS')
export const PRESENTER = createRequestTypes('PRESENTER')
export const CONFERENCES = createRequestTypes('CONFERENCES')
export const CONFERENCE = createRequestTypes('CONFERENCE')
export const SPONSORS = createRequestTypes('SPONSORS')
export const SPONSOR = createRequestTypes('SPONSOR')
export const SERIES = createRequestTypes('SERIES')
export const SERIE = createRequestTypes('SERIE')
export const TOPICS = createRequestTypes('TOPICS')
export const TOPIC = createRequestTypes('TOPIC')

export const LOAD_BIBLE_BOOKS = 'LOAD_BIBLE_BOOKS'
export const LOAD_BIBLE_CHAPTERS = 'LOAD_BIBLE_CHAPTERS'
export const LOAD_NEW_RECORDINGS = 'LOAD_NEW_RECORDINGS'
export const LOAD_TRENDING_RECORDINGS = 'LOAD_TRENDING_RECORDINGS'
export const LOAD_FEATURED_RECORDINGS = 'LOAD_FEATURED_RECORDINGS'
export const LOAD_BOOKS = 'LOAD_BOOKS'
export const LOAD_BOOK = 'LOAD_BOOK'
export const LOAD_STORIES = 'LOAD_STORIES'
export const LOAD_STORY = 'LOAD_STORY'
export const LOAD_PRESENTERS = 'LOAD_PRESENTERS'
export const LOAD_PRESENTER = 'LOAD_PRESENTER'
export const LOAD_CONFERENCES = 'LOAD_CONFERENCES'
export const LOAD_CONFERENCE = 'LOAD_CONFERENCE'
export const LOAD_SPONSORS = 'LOAD_SPONSORS'
export const LOAD_SPONSOR = 'LOAD_SPONSOR'
export const LOAD_SERIES = 'LOAD_SERIES'
export const LOAD_SERIE = 'LOAD_SERIE'
export const LOAD_TOPICS = 'LOAD_TOPICS'
export const LOAD_TOPIC = 'LOAD_TOPIC'

export const bibleBooks = {
  request: id => action(BIBLE_BOOKS.REQUEST, {id}),
  success: (id, response) => action(BIBLE_BOOKS.SUCCESS, {id, response}),
  refresh: (id, response) => action(BIBLE_BOOKS.REFRESH, {id, response}),
  failure: (id, error) => action(BIBLE_BOOKS.FAILURE, {id, error})
}

export const bibleChapters = {
  request: id => action(BIBLE_CHAPTERS.REQUEST, {id}),
  success: (id, response) => action(BIBLE_CHAPTERS.SUCCESS, {id, response}),
  refresh: (id, response) => action(BIBLE_CHAPTERS.REFRESH, {id, response}),
  failure: (id, error) => action(BIBLE_CHAPTERS.FAILURE, {id, error})
}

export const newRecordings = {
  request: id => action(NEW_RECORDINGS.REQUEST, {id}),
  success: (id, response) => action(NEW_RECORDINGS.SUCCESS, {id, response}),
  refresh: (id, response) => action(NEW_RECORDINGS.REFRESH, {id, response}),
  failure: (id, error) => action(NEW_RECORDINGS.FAILURE, {id, error})
}

export const trendingRecordings = {
  request: id => action(TRENDING_RECORDINGS.REQUEST, {id}),
  success: (id, response) => action(TRENDING_RECORDINGS.SUCCESS, {id, response}),
  refresh: (id, response) => action(TRENDING_RECORDINGS.REFRESH, {id, response}),
  failure: (id, error) => action(TRENDING_RECORDINGS.FAILURE, {id, error})
}

export const featuredRecordings = {
  request: id => action(FEATURED_RECORDINGS.REQUEST, {id}),
  success: (id, response) => action(FEATURED_RECORDINGS.SUCCESS, {id, response}),
  refresh: (id, response) => action(FEATURED_RECORDINGS.REFRESH, {id, response}),
  failure: (id, error) => action(FEATURED_RECORDINGS.FAILURE, {id, error})
}

export const books = {
  request: id => action(BOOKS.REQUEST, {id}),
  success: (id, response) => action(BOOKS.SUCCESS, {id, response}),
  refresh: (id, response) => action(BOOKS.REFRESH, {id, response}),
  failure: (id, error) => action(BOOKS.FAILURE, {id, error})
}

export const book = {
  request: id => action(BOOK.REQUEST, {id}),
  success: (id, response) => action(BOOK.SUCCESS, {id, response}),
  refresh: (id, response) => action(BOOK.REFRESH, {id, response}),
  failure: (id, error) => action(BOOK.FAILURE, {id, error})
}

export const stories = {
  request: id => action(STORIES.REQUEST, {id}),
  success: (id, response) => action(STORIES.SUCCESS, {id, response}),
  refresh: (id, response) => action(STORIES.REFRESH, {id, response}),
  failure: (id, error) => action(STORIES.FAILURE, {id, error})
}

export const story = {
  request: id => action(STORY.REQUEST, {id}),
  success: (id, response) => action(STORY.SUCCESS, {id, response}),
  refresh: (id, response) => action(STORY.REFRESH, {id, response}),
  failure: (id, error) => action(STORY.FAILURE, {id, error})
}

export const presenters = {
  request: id => action(PRESENTERS.REQUEST, {id}),
  success: (id, response) => action(PRESENTERS.SUCCESS, {id, response}),
  refresh: (id, response) => action(PRESENTERS.REFRESH, {id, response}),
  failure: (id, error) => action(PRESENTERS.FAILURE, {id, error})
}

export const presenter = {
  request: id => action(PRESENTER.REQUEST, {id}),
  success: (id, response) => action(PRESENTER.SUCCESS, {id, response}),
  refresh: (id, response) => action(PRESENTER.REFRESH, {id, response}),
  failure: (id, error) => action(PRESENTER.FAILURE, {id, error})
}

export const conferences = {
  request: id => action(CONFERENCES.REQUEST, {id}),
  success: (id, response) => action(CONFERENCES.SUCCESS, {id, response}),
  refresh: (id, response) => action(CONFERENCES.REFRESH, {id, response}),
  failure: (id, error) => action(CONFERENCES.FAILURE, {id, error})
}

export const conference = {
  request: id => action(CONFERENCE.REQUEST, {id}),
  success: (id, response) => action(CONFERENCE.SUCCESS, {id, response}),
  refresh: (id, response) => action(CONFERENCE.REFRESH, {id, response}),
  failure: (id, error) => action(CONFERENCE.FAILURE, {id, error})
}

export const sponsors = {
  request: id => action(SPONSORS.REQUEST, {id}),
  success: (id, response) => action(SPONSORS.SUCCESS, {id, response}),
  refresh: (id, response) => action(SPONSORS.REFRESH, {id, response}),
  failure: (id, error) => action(SPONSORS.FAILURE, {id, error})
}

export const sponsor = {
  request: id => action(SPONSOR.REQUEST, {id}),
  success: (id, response) => action(SPONSOR.SUCCESS, {id, response}),
  refresh: (id, response) => action(SPONSOR.REFRESH, {id, response}),
  failure: (id, error) => action(SPONSOR.FAILURE, {id, error})
}

export const series = {
  request: id => action(SERIES.REQUEST, {id}),
  success: (id, response) => action(SERIES.SUCCESS, {id, response}),
  refresh: (id, response) => action(SERIES.REFRESH, {id, response}),
  failure: (id, error) => action(SERIES.FAILURE, {id, error})
}

export const serie = {
  request: id => action(SERIE.REQUEST, {id}),
  success: (id, response) => action(SERIE.SUCCESS, {id, response}),
  refresh: (id, response) => action(SERIE.REFRESH, {id, response}),
  failure: (id, error) => action(SERIE.FAILURE, {id, error})
}

export const topics = {
  request: id => action(TOPICS.REQUEST, {id}),
  success: (id, response) => action(TOPICS.SUCCESS, {id, response}),
  refresh: (id, response) => action(TOPICS.REFRESH, {id, response}),
  failure: (id, error) => action(TOPICS.FAILURE, {id, error})
}

export const topic = {
  request: id => action(TOPIC.REQUEST, {id}),
  success: (id, response) => action(TOPIC.SUCCESS, {id, response}),
  refresh: (id, response) => action(TOPIC.REFRESH, {id, response}),
  failure: (id, error) => action(TOPIC.FAILURE, {id, error})
}

export const loadBibleBooks = (loadMore, refresh) => action(LOAD_BIBLE_BOOKS, {loadMore, refresh})
export const loadBibleChapters = (loadMore, refresh, testament, book) => action(LOAD_BIBLE_CHAPTERS, {loadMore, refresh, testament, book})
export const loadNewRecordings = (loadMore, refresh) => action(LOAD_NEW_RECORDINGS, {loadMore, refresh})
export const loadTrendingRecordings = (loadMore, refresh) => action(LOAD_TRENDING_RECORDINGS, {loadMore, refresh})
export const loadFeaturedRecordings = (loadMore, refresh) => action(LOAD_FEATURED_RECORDINGS, {loadMore, refresh})
export const loadBooks = (loadMore, refresh) => action(LOAD_BOOKS, {loadMore, refresh})
export const loadBook = (loadMore, refresh, url) => action(LOAD_BOOK, {loadMore, refresh, url})
export const loadStories = (loadMore, refresh) => action(LOAD_STORIES, {loadMore, refresh})
export const loadStory = (loadMore, refresh, url) => action(LOAD_STORY, {loadMore, refresh, url})
export const loadPresenters = (loadMore, refresh) => action(LOAD_PRESENTERS, {loadMore, refresh})
export const loadPresenter = (loadMore, refresh, url) => action(LOAD_PRESENTER, {loadMore, refresh, url})
export const loadConferences = (loadMore, refresh) => action(LOAD_CONFERENCES, {loadMore, refresh})
export const loadConference = (loadMore, refresh, url) => action(LOAD_CONFERENCE, {loadMore, refresh, url})
export const loadSponsors = (loadMore, refresh) => action(LOAD_SPONSORS, {loadMore, refresh})
export const loadSponsor = (loadMore, refresh, url) => action(LOAD_SPONSOR, {loadMore, refresh, url})
export const loadSeries = (loadMore, refresh) => action(LOAD_SERIES, {loadMore, refresh})
export const loadSerie = (loadMore, refresh, url) => action(LOAD_SERIE, {loadMore, refresh, url})
export const loadTopics = (loadMore, refresh) => action(LOAD_TOPICS, {loadMore, refresh})
export const loadTopic = (loadMore, refresh, url) => action(LOAD_TOPIC, {loadMore, refresh, url})

export const changeLanguage = language => action(CHANGE_LANGUAGE, {language})

export const playerState = state => action(PLAYER_STATE, {state})
export const playbackInit = () => action(PLAYBACK_INIT)
export const playbackTrackId = trackId => action(PLAYBACK_TRACK_ID, {trackId})
export const playbackTracks = tracks => action(PLAYBACK_TRACKS, {tracks})

export const resetAndPlayTrack = (tracks, id) => action(RESET_AND_PLAY_TRACK, {tracks, id})
export const playPause = () => action(PLAY_PAUSE)

export const skipToPrevious = () => action(SKIP_TO_PREVIOUS)
export const skipToNext = () => action(SKIP_TO_NEXT)
export const replay = () => action(REPLAY)
export const forward = () => action(FORWARD)
export const download = (item, downloadPath, downloadUrl, fileName, bitRate) => action(DOWNLOAD, {item, downloadPath, downloadUrl, fileName, bitRate})
export const setRate = rate => action(SET_RATE, {rate})
export const trackInitialized = (track) => action(TRACK_INITIALIZED, {track})
export const playbackRate = rate => action(PLAYBACK_RATE, {rate})
export const playbackPosition = position => action(PLAYBACK_POSITION, {position})

export const setBibleVersion = version => action(SET_BIBLE_VERSION, {version})
export const bibleVersion = version => action(BIBLE_VERSION, {version})
export const bibleBook = (testament, book) => action(BIBLE_BOOK, {testament, book})
export const bibleChapter = chapter => action(BIBLE_CHAPTER, {chapter})

export const ADD_TO_DOWNLOADS_QUEUE = 'ADD_TO_DOWNLOADS_QUEUE'
export const REMOVE_FROM_DOWNLOADS_QUEUE = 'REMOVE_FROM_DOWNLOADS_QUEUE'
export const SET_DOWNLOADING = 'SET_DOWNLOADING'
export const DOWNLOAD_PROGRESS = 'DOWNLOAD_PROGRESS'
export const addToDownloadsQueue = (item) => action(ADD_TO_DOWNLOADS_QUEUE, {item})
export const removeFromDownloadsQueue = (item) => action(REMOVE_FROM_DOWNLOADS_QUEUE, {item})
export const setDownloading = (downloading) => action(SET_DOWNLOADING, {downloading})
export const downloadProgress = (item, progress) => action(DOWNLOAD_PROGRESS, {item, progress})

const SET = 'SET'
const ADD = 'ADD'
const REMOVE = 'REMOVE'

function createMyListsTypes(base) {
  return [SET, ADD, REMOVE].reduce((acc, type) => {
    acc[type] = `${base}_${type}`
    return acc
  }, {})
}

export const DOWNLOADS = createMyListsTypes('DOWNLOADS')
export const downloads = {
  set: items => action(DOWNLOADS.SET, {items}),
  add: items => action(DOWNLOADS.ADD, {items}),
  remove: item => action(DOWNLOADS.REMOVE, {item})
}

export const REMOVE_DOWNLOAD = 'REMOVE_DOWNLOAD'
export const removeDownload = item => action(REMOVE_DOWNLOAD, {item})

export const FAVORITES = createMyListsTypes('FAVORITES')
export const favorites = {
  set: items => action(FAVORITES.SET, {items}),
  add: items => action(FAVORITES.ADD, {items}),
  remove: item => action(FAVORITES.REMOVE, {item})
}

export const PLAYLISTS = createMyListsTypes('PLAYLISTS')
export const playlists = {
  set: items => action(PLAYLISTS.SET, {items}),
  add: items => action(PLAYLISTS.ADD, {items}),
  remove: item => action(PLAYLISTS.REMOVE, {item})
}

export const PLAYLIST_ITEMS = createMyListsTypes('PLAYLIST_ITEMS')
export const playlistsItems = {
  set: items => action(PLAYLIST_ITEMS.SET, {items}),
  add: items => action(PLAYLIST_ITEMS.ADD, {items}),
  remove: item => action(PLAYLIST_ITEMS.REMOVE, {item})
}

export const HISTORY = createMyListsTypes('HISTORY')
export const history = {
  set: items => action(HISTORY.SET, {items}),
  add: items => action(HISTORY.ADD, {items}),
  remove: item => action(HISTORY.REMOVE, {item})
}

export const AUTOPLAY = 'AUTOPLAY'
export const setAutoPlay = autoPlay => action(AUTOPLAY, {autoPlay})

export const USER = 'USER'
export const setUser = user => action(USER, {user})

export const SYNC_FAVORITES = 'SYNC_FAVORITES'
export const syncFavorites = () => action(SYNC_FAVORITES)
export const ADD_FAVORITE = 'ADD_FAVORITE'
export const addFavorite = item => action(ADD_FAVORITE, {item})
export const REMOVE_FAVORITE = 'REMOVE_FAVORITE'
export const removeFavorite = id => action(REMOVE_FAVORITE, {id})

export const SYNC_PLAYLISTS = 'SYNC_PLAYLISTS'
export const syncPlaylists = () => action(SYNC_PLAYLISTS)
export const ADD_PLAYLIST = 'ADD_PLAYLIST'
export const addPlaylist = item => action(ADD_PLAYLIST, {item})
export const REMOVE_PLAYLIST = 'REMOVE_PLAYLIST'
export const removePlaylist = item => action(REMOVE_PLAYLIST, {item})

export const SYNC_PLAYLIST_ITEMS = 'SYNC_PLAYLIST_ITEMS'
export const syncPlaylistItems = (playlistId) => action(SYNC_PLAYLIST_ITEMS, {playlistId})
export const ADD_PLAYLIST_ITEM = 'ADD_PLAYLIST_ITEM'
export const addPlaylistItem = (playlistId, item) => action(ADD_PLAYLIST_ITEM, {playlistId, item})
export const REMOVE_PLAYLIST_ITEM = 'REMOVE_PLAYLIST_ITEM'
export const removePlaylistItem = (playlistId, id) => action(REMOVE_PLAYLIST_ITEM, {playlistId, id})

export const PLAY_VIDEO = 'PLAY_VIDEO'
export const playVideo = item => action(PLAY_VIDEO, {item})

export const LOG_OUT = 'LOG_OUT'
export const logOut = item => action(LOG_OUT)
