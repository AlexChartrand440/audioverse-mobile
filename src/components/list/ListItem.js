import React from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity, View, Text, Image, Platform, StyleSheet } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'

const ListItem = ({ style, avatar, icon, title, subtitle, rightElement, chevron, chevronColor, onPress }) => (
  <View>
    <PadView style={[styles.container, style]}>
      {renderIcon(icon)}
      {renderAvatar(avatar)}
      {(title || subtitle) && (
        <TouchableOpacity style={styles.contentContainer} onPress={onPress}>
          {renderNode(title, null, [styles.title])}
          {renderNode(subtitle, null, [styles.subtitle])}
        </TouchableOpacity>
      )}
      {renderNode(rightElement)}
      {chevron && <Icon name="chevron-right" size={26} color={chevronColor} onPress={onPress} />}
    </PadView>
    <View style={styles.divider} />
  </View>
)

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
    padding: Platform.OS === 'ios' ? 10 : 8
  },
  avatar: {
    width: 50,
    height: 50,
    borderRadius: 25
  },
  contentContainer: {
    flex: 1,
    justifyContent: 'center'
  },
  title: {
    fontSize: Platform.OS === 'ios' ? 17 : 16
  },
  subtitle: {
    fontSize: Platform.OS === 'ios' ? 15 : 14,
    color: '#212121'
  },
  divider: {
    // darker color if hairlineWidth is not thin enough
    backgroundColor: StyleSheet.hairlineWidth < 1 ? '#BCBBC1' : 'rgba(0, 0, 0, 0.12)',
    height: StyleSheet.hairlineWidth
  }
})

const elementOrObject = PropTypes.oneOfType([
  PropTypes.element,
  PropTypes.object
])

const elementOrString = PropTypes.oneOfType([
  PropTypes.element,
  PropTypes.string
])

ListItem.propTypes = {
  avatar: elementOrObject,
  icon: elementOrObject,
  title: elementOrString,
  subtitle: elementOrString,
  rightElement: PropTypes.element,
  chevron: PropTypes.bool,
  chevronColor: PropTypes.string,
  onPress: PropTypes.func
}

ListItem.defaultProps = {
  chevronColor: '#D1D1D6'
}

const PadView = ({ children, pad = 16, ...props }) => {
  const childrens = React.Children.toArray(children)
  const length = childrens.length
  return (
    <View {...props}>
      {React.Children.map(
        childrens,
        (child, index) =>
          child && [child, index !== length -1 && <View width={pad} />]
      )}
    </View>
  )
}

const renderAvatar = content =>
  content == null ? null : React.isValidElement(content) ? (
    content
  ) : (
    <Image {...content} source={content.source && content.source.toString().startsWith('http') ? {uri: content.source} : content.source} style={[styles.avatar, content.style]} />
  )

const renderIcon = content =>
  content == null ? null : React.isValidElement(content) ? (
    content
  ) : (
    <Icon size={24} color="rgba(0, 0, 0, 0.40)" {...content} />
  )

const renderNode = (content, props, style) =>
  content == null ? null : React.isValidElement(content) ? (
    content
  ) : (
    <Text ellipsizeMode={'tail'} numberOfLines={1} {...props} style={[style, props && props.style]}>
      {content}
    </Text>
  )

export default ListItem
