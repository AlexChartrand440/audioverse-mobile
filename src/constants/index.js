import Endpoints from './endpoints'
import ContentTypes from './contentTypes'
import Dirs from './dirs'
import Bitrates from './bitrates'

export {
  Endpoints,
  ContentTypes,
  Dirs,
  Bitrates
}
