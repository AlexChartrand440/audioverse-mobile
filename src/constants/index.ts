import Bibles from './bibles'
import Bitrates from './bitrates'
import ContentTypes from './contentTypes'
import Dirs from './dirs'
import Endpoints from './endpoints'
import Queries from './queries'

export {
  Endpoints,
  ContentTypes,
  Dirs,
  Bitrates,
  Bibles,
  Queries,
}
