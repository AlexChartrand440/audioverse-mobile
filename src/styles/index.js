import { StyleSheet } from 'react-native'

export const GlobalStyles = StyleSheet.create({
  header: {
    backgroundColor: '#E53935',
    elevation: 0,
    borderBottomWidth: 0
  },
  tab: {
    backgroundColor: '#E53935'
  },
  tabIndicator: {
    backgroundColor: '#FFFFFF'
  }
})

export const headerTintColor = '#FFFFFF'
