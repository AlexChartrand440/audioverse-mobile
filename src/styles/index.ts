import { StyleSheet } from 'react-native'
import image from '../../assets/default.png'

export const primaryColor = '#E53935'

export const GlobalStyles = StyleSheet.create({
  header: {
    backgroundColor: primaryColor,
    elevation: 0,
    borderBottomWidth: 0,
    shadowOpacity: 0,
  },
  tab: {
    backgroundColor: primaryColor
  },
  tabIndicator: {
    backgroundColor: '#FFFFFF'
  }
})

export const headerTintColor = '#FFFFFF'
export const defaultImage = image

const htmlTextStyle = {
  fontSize: 18,
  textAlign: 'center',
}
export const HTMLStyles =  {
  p: {
    ...htmlTextStyle,
    marginBottom: 20,
  },
  div: htmlTextStyle,
  li: {
    fontSize: 20,
  },
}
