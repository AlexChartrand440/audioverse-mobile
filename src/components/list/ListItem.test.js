import React from 'react'
import { Text } from 'react-native'
import ListItem from './ListItem'
import { defaultImage } from 'src/styles'

import renderer from 'react-test-renderer'

describe('ListItem component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ListItem />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with avatar', () => {
    const tree = renderer.create(<ListItem avatar={{source: defaultImage}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with icon', () => {
    const tree = renderer.create(<ListItem icon={{name: 'pocket'}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with icon component', () => {
    const tree = renderer.create(<ListItem icon={<Text>Icon</Text>} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with title and subtitle', () => {
    const tree = renderer.create(<ListItem title="title" subtitle="subtitle" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with right component', () => {
    const tree = renderer.create(<ListItem rightElement={<Text>Right component</Text>} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
