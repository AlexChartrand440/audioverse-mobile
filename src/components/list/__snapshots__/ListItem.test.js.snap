// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "padding": 10,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      }
    }
  />
</View>
`;

exports[`ListItem component renders correctly with avatar 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "padding": 10,
        },
        undefined,
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      }
    }
  />
</View>
`;

exports[`ListItem component renders correctly with icon 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "padding": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.40)",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î¦´
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      }
    }
  />
</View>
`;

exports[`ListItem component renders correctly with icon component 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "padding": 10,
        },
        undefined,
      ]
    }
  >
    <Text>
      Icon
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      }
    }
  />
</View>
`;

exports[`ListItem component renders correctly with right component 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "padding": 10,
        },
        undefined,
      ]
    }
  >
    <Text>
      Right component
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      }
    }
  />
</View>
`;

exports[`ListItem component renders correctly with title and subtitle 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "padding": 10,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Array [
              Object {
                "fontSize": 17,
              },
            ],
            null,
          ]
        }
      >
        title
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Array [
              Object {
                "color": "#212121",
                "fontSize": 15,
              },
            ],
            null,
          ]
        }
      >
        subtitle
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      }
    }
  />
</View>
`;
